scripts/oscal_normalize.py

```diff
@@ -793,6 +793,32 @@ def update_refs_per_file(classes):
     return classes
 
 
+def _strip_unrefed_files(file_class):
+    """
+    Strip unreferenced classes from each oscal file.
+
+    The generated oscal files include classes that aren't referenced within the file.
+    Those classes need to be removed early since they shouldn't be there at all.
+    A key problem class is SetParameter, but there are others.
+    This does a simple check of the file name being present at all in the body other classes.
+    It could instead be a token-based search but this seems sufficient.
+    """
+    dead_names = []
+    for c in file_class:
+        if c.name == 'Model':
+            continue
+        refd = False
+        for d in file_class:
+            if d.name in [c.name] + dead_names:
+                continue
+            if c.name in d.body_text:
+                refd = True
+                break
+        if not refd:
+            dead_names.append(c.name)
```

Comment:
This is rather a microoptimization, but this is of at least quadratic complexity. I can see two ways to improve it. One way to slightly improve it is to create a dictionary of `c` names that are already not "dead" and prevent algorithm to go to the second loop.

This is also can be done in linear time by creating a dictionary of all names, and then going through all body_texts once.

Reply:
Per discussion I think this is about as optimal as it can be since the list of classes has unique names.